### PR TITLE
Moved winhttp tests to WS2022 for WSL support

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Removed ACI proxy in favor of a localhost proxy.

Moved some tests to WS2022 and configure a squid proxy on localhost running under WSL..

Also reformatted platform matrixes to be consistent.

Also updated the live test platform matrix to include the OS name in the "StaticConfigs" section rather than in the "TargetPlatform" sections

## Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [ ] Related issue listed
- [X] Comments in source
- [X] No typos
- [X] Update changelog
- [X] Not work-in-progress
- [X] External references or docs updated
- [X] Self review of PR done
- [X] Any breaking changes?

### CODE REVIEW

File: eng/scripts/Get-BinarySizes.ps1 [hunk 1]
```diff
@@ -41,13 +41,16 @@ function setEnvVar($key, $value) {
 }
 
 function getTargetOs {
-    if ($OsVMImage.StartsWith('macOS')) {
+    if ($OsVMImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
```
Comment (benbp) [inline eng/scripts/Get-BinarySizes.ps1:44]: I assume #3900 will go in first?

**Review by benbp — APPROVED** [review]

File: eng/scripts/Get-BinarySizes.ps1 (hunk 1 above)
Comment (LarryOsterman) [reply eng/scripts/Get-BinarySizes.ps1:44]: They're going into two different branches - #3900 is going into main, this is going into /feature/websockets. Basically I ran into #3899 while I was working on this fix and I fixed it here. I realized this was affecting our nightly builds so I pulled some pieces of this PR into main (the reformatting of the platform matrix JSON files and this fix).

I pulled the platform matrix reformatting to make it easier to ultimately merge the /feature/websockets branch with main.

**Pull request merged**